pkg/apis/velero/v1/backup.go

```diff
@@ -273,6 +273,26 @@ type BackupStatus struct {
 	// file in object storage.
 	// +optional
 	Errors int `json:"errors,omitempty"`
+
+	// Progress contains information about the backup's execution progress. Note
+	// that this information is best-effort only -- if Velero fails to update it
+	// during a backup for any reason, it may be inaccurate/stale.
+	Progress *BackupProgress `json:"progress,omitempty"`
```

Comment:
A note for us - this will require users to install updated CRDs with v1.4.0. We'll need to put that in our docs.

Reply:
Yeah, good callout, I was thinking about that. I know we have the `--crds-only` option so I think we can instruct users to use that, dump it to YAML, and `kubectl apply` the output. I think we've done this before.

Reply:
What would be the failure if someone doesn't upgrade their CRDs?
May be out of scope for this PR, but we might want to detect that and fail with a "upgrade your CRDs" message than a crashloop

Reply:
There was this [older one that we had](https://github.com/vmware-tanzu/velero/blob/master/pkg/cmd/server/server.go#L408), but it's clearly out of date now as it talks about the YAML files we had in-tree. Plus, the condition that it looks for is only if the API group exists, no that relevant fields are found.

In general though, I'm 👍  to some sort of utility function that does a field-level verification for CRD compat. Ideally we'd be doing SemVer and do a patch release with added fields to make it really fast.

Reply:
We do compile the CRD manifests in (for the purposes of `velero install`), so I imagine we could use those to diff against what's in-cluster. I agree that we can tackle that in a separate issue/PR, especially since this feature has reasonable behavior IMHO if the CRDs are out of date (see 1 minute ago).

Reply:
#2509 captures the upgrade instructions, which is all that's needed for this PR.